app/controllers/v1/sessions_controller.rb

```diff
@@ -130,70 +129,69 @@ def user_login(saml_response)
       @current_user, @session_object = user_session_form.persist
       set_cookies
       after_login_actions
-      helper = url_service(user_session_form.saml_uuid)
-      if helper.should_uplevel?
-        render_login('verify', user_session_form.saml_uuid)
+      if url_service.should_uplevel?
+        render_login('verify')
       else
-        redirect_to helper.login_redirect_url
-        login_stats(:success, saml_response)
+        redirect_to url_service.login_redirect_url
+        login_stats(:success)
       end
     end
 
-    def render_login(type, previous_saml_uuid = nil)
-      force = (type != 'custom')
-      helper = url_service(previous_saml_uuid, force)
-      login_url, post_params = login_params(type, helper)
+    def render_login(type)
+      login_url, post_params = login_params(type)
+      tracker = url_service.tracker
       renderer = ActionController::Base.renderer
       renderer.controller.prepend_view_path(Rails.root.join('lib', 'saml', 'templates'))
       result = renderer.render template: 'sso_post_form',
                                locals: {
                                  url: login_url,
                                  params: post_params,
-                                 id: helper.tracker.uuid,
-                                 authn: helper.tracker.payload_attr(:authn_context),
-                                 type: helper.tracker.payload_attr(:type),
+                                 id: tracker.uuid,
+                                 authn: tracker.payload_attr(:authn_context),
+                                 type: tracker.payload_attr(:type),
                                  sentrydsn: Settings.sentry.dsn
                                },
                                format: :html
       render body: result, content_type: 'text/html'
-      saml_request_stats(helper.tracker)
+      saml_request_stats
     end
 
     # rubocop:disable Metrics/CyclomaticComplexity
-    def login_params(type, helper)
+    def login_params(type)
       raise Common::Exceptions::RoutingError, type unless REDIRECT_URLS.include?(type)
 
       case type
       when 'signup'
-        helper.signup_url
+        url_service.signup_url
       when 'mhv'
-        helper.mhv_url
+        url_service.mhv_url
       when 'dslogon'
-        helper.dslogon_url
+        url_service.dslogon_url
       when 'idme'
-        helper.idme_url
+        url_service.idme_url
       when 'mfa'
-        helper.mfa_url
+        url_service.mfa_url
       when 'verify'
-        helper.verify_url
+        url_service.verify_url
       when 'custom'
         raise Common::Exceptions::ParameterMissing, 'authn' if params[:authn].blank?
 
-        helper.custom_url params[:authn]
+        url_service(false).custom_url params[:authn]
       end
     end
     # rubocop:enable Metrics/CyclomaticComplexity
 
-    def saml_request_stats(tracker)
+    def saml_request_stats
+      t = url_service.tracker
```

Comment:
is there a reason this was shortened to `t` instead of being `tracker`?

Reply:
no good reason, I can revert